### PR TITLE
allow project admin to pause/resume flow trigger schedule

### PR DESCRIPTION
**Pull request by chengren311**
The thinking behind that is scheduled flow trigger needs to be paused if something goes south, e.x: the triggered flow is consuming too much azkaban resource. One way to stop the schedule is reupload a new project with new flow config with flow trigger turned off, but allowing the project admin to pause/resume schedule is apparently much more convenient option. 

Tested manually. Unit test will follow up.

### CODE REVIEW

File: azkaban-web-server/src/main/resources/azkaban/webapp/servlet/velocity/flowtriggerspage.vm [hunk 1]
```diff
@@ -130,6 +163,20 @@
                 </div>
               </div>
 
+              <td>${trigger.isPaused()}</td>
```
Comment (kunkun-tang) [inline azkaban-web-server/src/main/resources/azkaban/webapp/servlet/velocity/flowtriggerspage.vm]: Synced offline. We should also have a "pause" button inside Project Flow page.

File: azkaban-web-server/src/main/java/azkaban/webapp/servlet/FlowTriggerServlet.java [hunk 2]
```diff
@@ -75,21 +81,52 @@ private void ajaxFetchTrigger(final int projectId, final String flowId, final Se
...
         final int projectId = getIntParam(req, "projectId");
         final String flowId = getParam(req, "flowId");
         ajaxFetchTrigger(projectId, flowId, session, ret);
-        if (ret != null) {
-          this.writeJSON(resp, ret);
+      }
+    } else if (ajaxName.equals("pauseTrigger") || ajaxName.equals("resumeTrigger")) {
+      if (checkProjectIdAndFlowId(req)) {
+        final int projectId = getIntParam(req, "projectId");
+        final String flowId = getParam(req, "flowId");
```
Comment (kunkun-tang) [inline azkaban-web-server/src/main/java/azkaban/webapp/servlet/FlowTriggerServlet.java]: Please also check if "flowID" is valid or not.

File: azkaban-web-server/src/main/java/azkaban/webapp/servlet/FlowTriggerServlet.java [hunk 3]
```diff
@@ -75,21 +81,52 @@ private void ajaxFetchTrigger(final int projectId, final String flowId, final Se
...
+          ret.put("error", "please specify a valid project id");
+        }
+        else if (!hasPermission(project, session.getUser(), Type.ADMIN)) {
+          ret.put("error", "Permission denied. Need ADMIN access.");
+        }
+        else {
+          try {
+            if(ajaxName.equals("pauseTrigger")) {
+              this.scheduler.pauseFlowTrigger(projectId, flowId);
+            }
```
Comment (kunkun-tang) [inline azkaban-web-server/src/main/java/azkaban/webapp/servlet/FlowTriggerServlet.java]: format issue. "else" should be placed in the line above

File: azkaban-web-server/src/main/java/azkaban/scheduler/QuartzScheduler.java [hunk 4]
```diff
@@ -111,6 +113,43 @@ public void shutdown() {
     }
   }
 
+  /**
+   * pause a job given the groupname. since pausing request might be issued concurrently,
+   * so synchronized is added to ensure thread safety.
+   */
+  public synchronized void pauseJob(final String groupName) throws SchedulerException {
```
[the remark this answers is not in the document]
Comment (chengren311) [reply azkaban-web-server/src/main/java/azkaban/scheduler/QuartzScheduler.java]: agree, will do it in another PR.

File: azkaban-web-server/src/main/resources/azkaban/webapp/servlet/velocity/flowtriggerspage.vm (hunk 1 above)
Comment (chengren311) [reply azkaban-web-server/src/main/resources/azkaban/webapp/servlet/velocity/flowtriggerspage.vm]: add it as todo.

File: azkaban-web-server/src/main/java/azkaban/webapp/servlet/FlowTriggerServlet.java (hunk 2 above)
Comment (chengren311) [reply azkaban-web-server/src/main/java/azkaban/webapp/servlet/FlowTriggerServlet.java]: updated.

**Pull request merged**